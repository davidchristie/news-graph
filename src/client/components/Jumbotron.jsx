import React, { PropTypes } from 'react'

export default class Jumbotron extends React.Component {
  render () {
    const children = this.props.children
    const title = this.props.title
    const lead = this.props.lead
    const text = this.props.text
    return (
      <div>
        <div className='jumbotron'>
          <h1 className='display-3'>{title}</h1>
          <p className='lead'>{lead}</p>
          <hr className='my-4' />
          <p>{text}</p>
          {children}
        </div>
      </div>
    )
  }
}

Jumbotron.propTypes = {
  children: PropTypes.element,
  lead: PropTypes.string,
  text: PropTypes.string,
  title: PropTypes.string
}
