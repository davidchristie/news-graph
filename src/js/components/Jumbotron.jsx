import React, { PropTypes } from 'react'

export default class Jumbotron extends React.Component {
  render() {
    const children = this.props.children
    const title = this.props.title
    return (
      <div>
        <div className="jumbotron">
          <h1 className="display-3">{title}</h1>
          <p className="lead">This is a simple hero unit, a simple jumbotron-style component for calling extra attention to featured content or information.</p>
          <hr className="my-4" />
          <p>It uses utility classes for typography and spacing to space content out within the larger container.</p>
          {children}
        </div>
      </div>
    )
  }
}

Jumbotron.propTypes = {
  children: PropTypes.element,
  title: PropTypes.string.isRequired
}
