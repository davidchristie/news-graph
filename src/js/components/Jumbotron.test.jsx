/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import Jumbotron from './Jumbotron'

test('Jumbotron renders correctly', () => {
  const tree = renderer.create(
    <Jumbotron title="jumbotron" />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
