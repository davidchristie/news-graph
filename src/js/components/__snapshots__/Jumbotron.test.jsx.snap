exports[`test Jumbotron renders correctly 1`] = `
<div>
  <div
    className="jumbotron">
    <h1
      className="display-3">
      jumbotron
    </h1>
    <p
      className="lead">
      This is a simple hero unit, a simple jumbotron-style component for calling extra attention to featured content or information.
    </p>
    <hr
      className="my-4" />
    <p>
      It uses utility classes for typography and spacing to space content out within the larger container.
    </p>
  </div>
</div>
`;
